src/hubdata/create_target_data_schema.py

```diff
@@ -0,0 +1,121 @@
+import json
+from pathlib import Path
+
+import pyarrow as pa
+
+from hubdata import HubConnection, connect_hub
+from hubdata.create_hub_schema import _pa_type_for_hub_type
+
+
+def create_timeseries_schema(hub_path: str | Path) -> pa.schema:
+    """
+    Top-level function for creating a time-series schema for the passed `hub_path`.
+
+    :param hub_path: str (for local file system hubs or cloud based ones) or Path (local file systems only) pointing to
+        a hub's root directory. it is passed to https://arrow.apache.org/docs/python/generated/pyarrow.fs.FileSystem.html#pyarrow.fs.FileSystem.from_uri
+        From that page: Recognized URI schemes are “file”, “mock”, “s3fs”, “gs”, “gcs”, “hdfs” and “viewfs”. In
+        addition, the argument can be a local path, either a pathlib.Path object or a str. NB: Passing a local path as a
+        str requires an ABSOLUTE path, but passing the hub as a Path can be a relative path.
+    :return: a `pyarrow.Schema` for the passed `hub_path`
+    :raise: RuntimeError if `hub_path` has no `hub-config/target-data.json` file
+    """
+    hub_conn = connect_hub(hub_path)
+    target_data = _target_data_json(hub_conn)  # try to open hub-config/target-data.json
+    col_name_to_pa_type = _col_name_to_pa_type_for_target_data(hub_conn.schema, target_data, True)
+    return pa.schema(col_name_to_pa_type)
+
+
+def create_oracle_output_schema(hub_path: str | Path) -> pa.schema:
+    """
+    Top-level function for creating a oracle-output schema for the passed `hub_path`.
+
+    :param hub_path: str (for local file system hubs or cloud based ones) or Path (local file systems only) pointing to
+        a hub's root directory. it is passed to https://arrow.apache.org/docs/python/generated/pyarrow.fs.FileSystem.html#pyarrow.fs.FileSystem.from_uri
+        From that page: Recognized URI schemes are “file”, “mock”, “s3fs”, “gs”, “gcs”, “hdfs” and “viewfs”. In
+        addition, the argument can be a local path, either a pathlib.Path object or a str. NB: Passing a local path as a
+        str requires an ABSOLUTE path, but passing the hub as a Path can be a relative path.
+    :return: a `pyarrow.Schema` for the passed `hub_path`
+    :raise: RuntimeError if `hub_path` has no `hub-config/target-data.json` file
+    """
+    hub_conn = connect_hub(hub_path)
+    target_data = _target_data_json(hub_conn)  # try to open hub-config/target-data.json
+    col_name_to_pa_type = _col_name_to_pa_type_for_target_data(hub_conn.schema, target_data, False)
+    return pa.schema(col_name_to_pa_type)
+
+
+def _target_data_json(hub_conn: HubConnection) -> dict:
```

Comment:
As I understand, the function will fail as it's missing the target-data.json meaning we cannot load the data without it. I think in the R package version, we allow hubs to not have the target-data.json and in this case infer the information from the data itself. Do we want the same behavior here? (and if so, do we want in this version or a future version)?